src/util/props.js

```diff
@@ -0,0 +1,46 @@
+
+import { warn } from './warn'
+
+export function resolveProps (route, component, config) {
+  switch (typeof config) {
+
+    case 'undefined':
+      return
+
+    case 'object':
+      return config
+
+    case 'function':
+      return config(route)
+
+    case 'boolean':
+      if (!config) {
+        return
+      }
+      if (!component.props) {
+        return
+      }
+      const props = {}
+      for (const prop in route.params) {
+        const value = route.params[prop]
+        if (hasProp(component, prop)) {
+          const propType = component.props[prop].type
```

Comment:
Maybe in that case, the user should provide a function and convert types by himself. This way we avoid the implicit behavior.